### DISCUSSION TITLE
Documentation about (global) state and variables in general?

### DISCUSSION THREAD

**Discussion by matthiasautrata** [writer/writer-framework#80] [Ideas]
I found references to attributes in the global state dictionary in various parts of the documentation. Would it be possible to have a section in the documentation that explains how this works and what entries are "pre-defined/expected?" It would be nice to have a single place where also things like "name starts with an `_` like `state["_my_private_state_var"]` could be found in one place.

**Comment by matthiasautrata** [comment]
Maybe it would also be helpful to explain when/where global variables are agreeable or not. I naively set a variable `the_current_flag = False` and modified that in various places only to find out that more often than not it shows up as reset to `False` again (automagically). I then changed the variable to a state-variable and things started to work fine.

I am sure it is my lack of understanding of the control flow in Streamsync.

**Comment by ramedina86** [comment]
Global variables are reset to its original definition when the programme is re-run. For example, when clicking "Save and Run" or saving the code (which triggers a re-run).

A section on [Application State](https://www.streamsync.cloud/application-state.html) is available in the documentation site. Have you read that one?

There are no predefined nor expected entries in state, you just add the ones you need.

**Comment by matthiasautrata** [comment]
Mea culpa. I did read that section indeed and find it very helpful. There are various other references to methods on state and I mixed up state variables and state methods. (e.g., `state["counter"] += 1` makes sense. Then again, `state.add_notification(...)`  and others also make sense. They just got mixed up during reading/learning, somehow.
